lib/go-tc/deliveryservice_ssl_keys.go

```diff
@@ -93,6 +93,15 @@ type DeliveryServiceSSLKeysV15 struct {
 	Expiration time.Time `json:"expiration,omitempty"`
 }
 
+// SSLKeyExpirationInformation contains information about an SSL key's expiration.
+type SSLKeyExpirationInformation struct {
+	DeliveryService string    `json:"deliveryservice"`
+	Cdn             string    `json:"cdn"`
+	Provider        *string   `json:"provider"`
```

Comment:
This is still a reference, and the underlying database allows null values for `provider`, but in practice they're all empty strings:
```plpgsql
$ SELECT deliveryservice, cdn, provider, expiration FROM sslkey WHERE provider IS NULL;
 deliveryservice | cdn | provider | expiration 
-----------------+-----+----------+------------
(0 rows)

$ SELECT deliveryservice, cdn, provider, expiration FROM sslkey WHERE provider = '';
 deliveryservice |     cdn      | provider |       expiration       
-----------------+--------------+----------+------------------------
 demo1           | CDN-in-a-Box |          | 2022-03-17 19:10:49+00
 demo1           | CDN-in-a-Box |          | 2022-03-17 19:10:49+00
 demo2           | CDN-in-a-Box |          | 2022-03-17 19:10:49+00
 demo2           | CDN-in-a-Box |          | 2022-03-17 19:10:49+00
(4 rows)
```

Should that be inserting `NULL` instead of `''`? Or maybe that column should have a `NOT NULL` constraint instead?

Reply:
i had it this way because when the DB migrate is run it has `NULL` in there and errored out if i had the `NOT NULL` requirement but after the `fill_expiration_and_provider` script is run then it should all be filled in or empty string. is there a better way to do that? im game to change it to all `NULL` but if theres a better way id prefer that stay as empty string

Reply:
i was hoping to not add another migration but thats the right way to go. updating it.